docs/button.md

```diff
@@ -120,9 +120,9 @@ export default App;
 
 Handler to be called when the user taps the button.
 
-| Type                                      |
-| ----------------------------------------- |
-| ({nativeEvent: [PressEvent](pressevent)}) |
+| Type                                           |
+| ---------------------------------------------- |
+| `md ({nativeEvent: [PressEvent](pressevent)})` |
```

Comment:
@cipolleschi @Simek @cortinico 

This new inline code syntax

```
`md ({nativeEvent: [PressEvent](pressevent)})`
```

Turns on a basic support for Markdown links inside

![CleanShot 2023-08-04 at 17 46 24](https://github.com/facebook/react-native-website/assets/749374/59d272b0-106b-4237-a7c1-71572c961fe6)

Can you validate this idea? 

IMHO this is an UX improvement, considering it's code it should be an inline code block. The non-usage of links before was probably a technical limitation that this new solution enables. 

But this requires to teach technical writers to use the <code>`md </code> prefix to turn it on. Considering the current implementation, I prefer to keep it opt-in instead of applied everywhere by default.

---

Review links:
- Before: https://reactnative.dev/docs/next/button#requiredonpress
- After: https://deploy-preview-3807--react-native.netlify.app/docs/next/button#requiredonpress

Reply:
I like it. I don't think that it will be an issue: if a technical writer wants to achieve the same result in these pages, it will likely go here and see how that was achieved and then it will copy the syntax!